### PR TITLE
Migrate to Legacy Builtin NDB Library

### PR DESCRIPTION
**Pull request by phil-lopreiato**
Towards https://github.com/the-blue-alliance/the-blue-alliance/issues/3995

This will simplify a few things, mainly the fact we can just have native interop between old/new apps without having to do a bunch of hackery to get datastore seriliazation/memcache to agree on a data format.

The couple things still remaining:
 - [x] figure out if we can still use the remote api locally. Answer: no, because adding `googledatastore` creates dependency hell. We can file an upstream issue for this.
 - [x] update docs

### CODE REVIEW

File: .pyre_configuration [hunk 1]
```diff
@@ -10,10 +10,7 @@
     {"site-package": "flask_cors"},
     {"site-package": "freezegun"},
     {"site-package": "google/appengine"},
-    {"site-package": "google/cloud/datastore_v1"},
```
Comment (ZachOrr) [inline .pyre_configuration:13]: We should also be able to remove things like `fakeredis` and `redis` here too, right? It also looks like I forgot to remove `rq`

File: requirements.txt [hunk 2]
```diff
@@ -6,15 +6,13 @@ flake8-tidy-imports
...
-InMemoryCloudDatastoreStub
 pre-commit
 pyre-check==0.9.8
 pytest
 pytest-cov
 pytest-custom_exit_code
 pytest-testmon
 PyYAML==6.0
-redis==3.5.3
 hiredis==2.0.0
```
Comment (ZachOrr) [inline requirements.txt:16]: We can remove `hiredis` here too

File: .pyre_configuration (hunk 1 above)
Comment (phil-lopreiato) [reply .pyre_configuration:13]: We still need redis momentarily (for raw memcache + flask handler caching).

Those will be easy enough to move out in separate PRs though

File: requirements.txt (hunk 2 above)
Comment (phil-lopreiato) [reply requirements.txt:16]: not yet, but soon!

Comment (ZachOrr) [reply requirements.txt:16]: Also `fakeredis`

**Pull request merged**